### ISSUE TITLE
[HOLD for payment 2023-09-27] [$500] Money in wallet badge is not center vertically

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#26308]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. go to settings
2. check the amount in badge
## Expected Result: 
money should be in center vertically
## Actual Result: 
money in badge is not center vertically

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop
 
**Version Number:** 1.3.59-1
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/b80a85ca-14cb-4d97-878a-dfd747c83629

**Expensify/Expensify Issue URL:**
**Issue reported by:** @gadhiyamanan 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692597419368009

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
![Snip - (100) New Expensify - Google Chrome](https://github.com/Expensify/App/assets/43996225/a1d02a07-f6c8-4e39-97c5-7592f96daf0e)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01ea8fed02b430b214</li>
        <li>Upwork Job ID: 1696985690595770368</li>
        <li>Last Price Increase: 2023-09-06</li>
<li>Automatic offers: </li>
<ul>
<li>ZhenjaHorbach | Contributor | 26600759</li>
<li>gadhiyamanan | Reporter | 26600761</li>
</ul></li>
    </ul>
</details>

**Comment by b4s36t4** [comment]
It is vertically aligned. We need to conside `$` symbol too.

**Comment by ZhenjaHorbach** [comment]
I think the problem is in lineHeight
But it only works on Safari for me
Chrome renders correctly lineHeight


For example, how Safari reacts to changes lineHeight
<img width="355" alt="Screenshot 2023-08-31 at 03 52 12" src="https://github.com/Expensify/App/assets/68128028/3c4ed104-29e1-43d4-8e03-437d0295f990">
<img width="355" alt="Screenshot 2023-08-31 at 03 52 17" src="https://github.com/Expensify/App/assets/68128028/b3fc3194-e062-4b2f-bc49-0e1bf0a83df7">

**Comment by ZhenjaHorbach** [comment]
Contributor details
Your Expensify account email: horbachyauhenipetrovich@gmail.com
Upwork Profile Link: https://www.upwork.com/freelancers/~01e2ef043508da9f2c

**Comment by anyongjitiger** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

The problem is that Money in wallet badge is not center, bottom is bigger than top.


### What is the root cause of that problem?

The root cause of the problem is  :
The line-height of the badgeText is smaller (16px) than the height of the badge (20px).

### What changes do you think we should make in order to solve the problem?

We can set the line-height of the badgeText and the height of the badge to be the same, which will center it.

![截屏2023-08-31 11 01 35](https://github.com/Expensify/App/assets/9712610/2a08a50c-626a-4f47-81bd-d0ca8c0bf228)


### What alternative solutions did you explore? (Optional)

None

**Comment by robertKozik** [comment]
Hi all! I'll address each proposal in separate points, as I have questions to both of them:
1. @ZhenjaHorbach — I'm worried about the regression on other badge instances - as we cannot be sure about the behaviour when we are not setting anything for lineHeight. Did you consider setting equal values for the lineHeight and fontSize? 
2. @anyongjitiger — Did you check your approach on the safari web? I'm getting a vibe that it's still not centered:
<img width="253" alt="image" src="https://github.com/Expensify/App/assets/61146594/e017ab7c-8b8a-471c-9e10-487e458a9a72">

**Comment by ZhenjaHorbach** [comment]
> Hi all! I'll address each proposal in separate points, as I have questions to both of them:
> 
> 1. @ZhenjaHorbach — I'm worried about the regression on other badge instances - as we cannot be sure about the behaviour when we are not setting anything for lineHeight. Did you consider setting equal values for the lineHeight and fontSize?
> 2. @anyongjitiger — Did you check your approach on the safari web? I'm getting a vibe that it's still not centered:
> 
> <img alt="image" width="253" src="https://user-images.githubusercontent.com/61146594/264614318-e017ab7c-8b8a-471c-9e10-487e458a9a72.png">

Hello )
Yes, I checked today morning, in fact this is the only correct acceptable value (fontSize value) )
And it turns out that in this case the available values are:
-styles without lineHeight;
-lineHeight which is equal to fontSize.

In other cases, bugs appear in Safari.

But maybe you're right, styles without lineHeight can lead to potential unpredictability in the future )

**Comment by anyongjitiger** [comment]
> Hi all! I'll address each proposal in separate points, as I have questions to both of them:
> 
> 1. @ZhenjaHorbach — I'm worried about the regression on other badge instances - as we cannot be sure about the behaviour when we are not setting anything for lineHeight. Did you consider setting equal values for the lineHeight and fontSize?
> 2. @anyongjitiger — Did you check your approach on the safari web? I'm getting a vibe that it's still not centered:
> 
> <img alt="image" width="253" src="https://user-images.githubusercontent.com/61146594/264614318-e017ab7c-8b8a-471c-9e10-487e458a9a72.png">

I just tested it on Safari and it does have display issues. It seems to be a font-related problem. When I disabled the ExpensifyNeue-Regular font in devtools, it displayed correctly. One possible solution could be adjusting the fonts, as mentioned in this answer: 
https://stackoverflow.com/a/15860322

**Comment by maddylewis** [comment]
@robertKozik - bump on proposal review - thx!

**Comment by robertKozik** [comment]
Hmmm thanks for your findings @anyongjitiger, if it's indeed the problem related with the font I think it's not a bug then. I'm heading OOO for today, but tomorrow morning I'll double check the alignment of the `$` sign, I think it can be vertically aligned

**Comment by robertKozik** [comment]
I've rechecked it, and it appears to be linked to the font `ExpenifyNeue-regular`, but that's not the sole factor. Even with the same font, there's a discrepancy between Safari and Chrome. Considering this, I'm inclined to agree with @ZhenjaHorbach's proposal for consistency across both browsers.

Selected Proposal: [proposal](https://github.com/Expensify/App/issues/26308#issuecomment-1700282042)
Contributor: @ZhenjaHorbach 

🎀 👀 🎀 C+ reviewed

**Comment by thienlnam** [comment]
Hmm, after looking at this again - I'm not sure this is the correct root cause.

From the original [proposal](https://github.com/Expensify/App/issues/26308#issuecomment-1700282042), it basically mentions that Chrome should already work as expected.

Then from this update - https://github.com/Expensify/App/pull/27137/files#r1324317248, if we're adjusting the line height for all of web then wouldn't it be incorrect now? 

cc @robertKozik could you let me know what you think about this?

**Comment by robertKozik** [comment]
There was no drawback to address whole web, not only the safari case, that's why I initially passed it in review. But after checking it out again, I agree that we should reduce scope to be safari-specific

**Comment by robertKozik** [comment]
It seemed there were no drawback to addressing the whole web rather than just the Safari case, which is why I initially passed it in review. However, upon reevaluation, I agree that we should reduce the scope to focus specifically on Safari.

**Comment by thienlnam** [comment]
Does not look like the issue mentioned here was fixed. https://github.com/Expensify/App/issues/27654#issuecomment-1726650591

**Comment by thienlnam** [comment]
cc @kavimuru Do you have more info about the reproducing this bug?

**Comment by robertKozik** [comment]
"Hi! It seems that the remaining discrepancy is likely due to the font, as mentioned [here](https://github.com/Expensify/App/issues/26308#issuecomment-1701975625). With the PR, we've addressed how Safari handles vertical centering. Previously, Safari behaved differently than Chrome, but now they're aligned."

before safari: 
<img width="95" alt="image" src="https://github.com/Expensify/App/assets/61146594/cd1efe67-3456-4dd9-a4c6-4ecd02997776">

after safari:
<img width="100" alt="image" src="https://github.com/Expensify/App/assets/61146594/6e2b9d86-36fa-41ef-a9f1-66611307630f">

**Comment by maddylewis** [comment]
## Payments
- [ ] Reporting bonus: @gadhiyamanan ($50 via Upwork)
- [ ] Contributor: @ZhenjaHorbach ($500 via Upwork)
- [ ] C+: @robertKozik ($500 via Upwork)

**Comment by maddylewis** [comment]
@robertKozik - can you review this checklist when you have a chance? thanks! https://github.com/Expensify/App/issues/26308#issuecomment-1727194081

**Comment by gadhiyamanan** [comment]
@maddylewis this is eligible for $250 because it was reported before 30 aug

**Comment by gadhiyamanan** [comment]
Issue is created before the announcement https://expensify.slack.com/archives/C01GTK53T8Q/p1693424078129649

**Comment by maddylewis** [comment]
@robertKozik - bump on this one ty! https://github.com/Expensify/App/issues/26308#issuecomment-1743040256

**Comment by maddylewis** [comment]
@robertKozik - bump https://github.com/Expensify/App/issues/26308#issuecomment-1727194081

**Comment by robertKozik** [comment]
> **BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:
> 
> * [X]  [@robertKozik] The PR that introduced the bug has been identified. Link to the PR: that was inconsistency between browsers, no particular PR introduced it
> * [X]  [@robertKozik] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: N/A
> * [X]  [@robertKozik] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: N/A
> * [X]  [@robertKozik] Determine if we should create a regression test for this bug. We don't need regression test for visual bugs
> * [X]  [@robertKozik] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again. N/A

tldr; I think we don't need regression test for this one as it's purely visual bug which don't interfere with any user flow

**Comment by maddylewis** [comment]
everyone is paid and a regression test isn't required since this is purely visual. closing!
